table/block_based/block.cc

```diff
@@ -394,6 +394,12 @@ void IndexBlockIter::Seek(const Slice& target) {
   bool ok = false;
   if (prefix_index_) {
     ok = PrefixSeek(target, &index);
+    if (!ok && !Valid()) {
+      // This is to let the caller to distinguish between non-existing prefix,
+      // and when key is larger than the last key, which both set Valid() to
+      // false.
```

Comment:
This comment probably is not totally correct as in prefix case, NotFound() may mean the key is larger than the last key while the prefix shows up too.

I think there might be some problem with that scenario too.